packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -588,6 +610,80 @@ export class NftDetectionController extends BaseController<
               ? elm.blockaidResult?.result_type === BlockaidResultType.Benign
               : true),
         );
+        // Retrieve collections from apiNfts
+        // contract and collection.id are equal for simple contract addresses; this is to exclude cases for shared contracts
+        const collections = apiNfts.reduce<string[]>((acc, currValue) => {
+          if (
+            !acc.includes(currValue.token.contract) &&
+            currValue.token.contract === currValue?.token?.collection?.id
```

Comment:
Pushing only when token.contract is equal collection.id which will exclude cases of shared contract addresses where collection.id can be equal "0xaddress...:1234:3445".
This can change in future to support these contracts once NFT-API also supports it 🙏